rust/src/database/managed_trie/store.rs

```diff
@@ -285,22 +427,42 @@ mod tests {
         let child_id = manager.insert(manager.make());
         let grandchild_id = manager.insert(manager.make());
 
+        let updates = KeyedUpdateBatch::from_key_value_pairs(&[(KEY, VALUE)]);
         thread::scope(|s| {
             s.spawn(|| {
                 let root_id_guard = root_id_lock.write().unwrap();
-                store(root_id_guard, &KEY, &VALUE, &*manager, &log).unwrap();
+                store(root_id_guard, &updates, &*manager, &log).unwrap();
             });
 
             manager.expect_write_access(root_id, vec![]);
             assert!(root_id_lock.try_read().is_err());
-            descend_into(&manager, root_id, child_id, KEY, 0);
-            assert!(root_id_lock.try_read().is_err());
-            descend_into(&manager, child_id, grandchild_id, KEY, 1);
+            descend_into(
+                &manager,
+                root_id,
+                None,
+                vec![DescendAction {
+                    id: child_id,
+                    updates: updates.clone(),
+                }],
+                &updates,
+                0,
+            );
             let _guard = spin_until_some(
```

Comment:
I guess the lock is now released one level earlier? So we don't need the second descend?